src/picotorrent/torrentdetails/torrenttrackerswidget.cpp

```diff
@@ -158,3 +167,16 @@ void TorrentTrackersWidget::onTrackerContextMenu(QPoint const& point)
     QObject::connect(menu, &QMenu::aboutToHide,
                      menu, &QMenu::deleteLater);
 }
+
+bool TorrentTrackersWidget::isTrackerUrl(QString url)
+{
+    for each (QString urlScheme in trackerUrlSchemes)
```

Comment:
In other places in PicoTorrent we use the simpler `for` iterator, like so,

```
for (QString& urlScheme : trackerUrlSchemes)
```

See if you can change to this 👍

Reply:
Thanks for your tip and positive feedback!

I have some questions out of curiosity.

From https://docs.microsoft.com/en-us/cpp/cpp/range-based-for-statement-cpp?view=vs-2019:
```c++
for( auto &y : x ) { // Type inference by reference.
	// Observes and/or modifies in-place. Preferred when modify is needed.
}

for( const auto &y : x ) { // Type inference by const reference.
	// Observes in-place. Preferred when no modify is needed.
}
```

My questions:
1. Do you want to use the `auto` keyword or be specific and use `QString`?
2. Do you want to use the `const` keyword, because we are not changing the value?

My suggestion:
```c++
for (const auto& urlScheme : trackerUrlSchemes)
```

Let me know what you want and I will make the change. Thanks!

Reply:
Hi! I think `auto` has its place when we're dealing with long type names, but for in this case, for `QString`, I'd rather stay away from it :) Add `const` - it's a good idea since we aren't modifying anything.